carbon-location.html

```diff
@@ -148,8 +148,8 @@
         }
       },
 
-      __computeRoutePath: function(path, hash, useHashAsPath) {
-        return useHashAsPath ? hash : path;
+      __computeRoutePath: function() {
+        return this.useHashAsPath ? this.__hash : this.__path;
```

Comment:
I think this deserves a closer look. My gut tells me that you're relying on the order of effects here. This becomes problematic especially in the init phase of Polymer. With the path changed upwards listener and this downwards stream you're building essentially an event loop. This is confusing for the reader and the system. I think these errors should actually guide you to try a different approach here, less procedural.